experimental/framework/fluid-static/src/serviceAudience.ts

```diff
@@ -12,67 +12,91 @@ import { IServiceAudience, IServiceAudienceEvents, IMember } from "./types";
 // Base class for providing audience information for sessions interacting with FluidContainer
 // This can be extended by different service-specific client packages to additional parameters to
 // the user and client details returned in IMember
-export class ServiceAudience
-  extends TypedEventEmitter<IServiceAudienceEvents<IMember>>
-  implements IServiceAudience<IMember> {
+export abstract class ServiceAudience<M extends IMember = IMember>
+  extends TypedEventEmitter<IServiceAudienceEvents<M>>
+  implements IServiceAudience<M> {
   protected readonly audience: IAudience;
 
+  /**
+   * Retain the most recent member list.  This is so we have more information about a member
+   * leaving the audience in the removeMember event.  It allows us to match the behavior of the
+   * addMember event where it only fires on a change to the members this class exposes (and would
+   * actually produce a change in what getMembers returns).  It also allows us to provide the
+   * client details in the event which makes it easier to find that client connection in a map
+   * keyed on the userId and not clientId.
+   * This map will always be up-to-date in a removeMember event because it is set once at
+   * construction and in every addMember event.
+   * It is mapped clientId to M to be better work with what the IAudience event provides
+   */
+  protected lastMembers: Map<string, M> = new Map();
```

Comment:
This could be simplified if the `IAudience` `removeMember` event also provides the `IClient` of the removed member (we will still probably want to convert that into an `M extends IMember` though).  Then we wouldn't need to keep track of the last member state to match a `userId` to the `clientId`

Reply:
Agreed, having the event itself provide the user that left as well as the clientId would be ideal instead of trying to keep an in-memory map updated based off of the events